### PR TITLE
Made the `UiVertex` and `UiMeta` fields publicly available.

### PR DESCRIPTION
**Pull request by silvestrpredko**
# Objective

It would be easier to customize the Bevy UI.
For example, I would like to create a plugin that will create rounded corners for the Bevy UI node.
With publicly available fields it will be easier to achieve.

## Solution

`UiVertex` become public and `UiMeta` fields too.

### CODE REVIEW

File: crates/bevy_ui/src/render/mod.rs [hunk 1]
```diff
@@ -600,7 +600,7 @@ pub fn extract_text_uinodes(
 
 #[repr(C)]
 #[derive(Copy, Clone, Pod, Zeroable)]
-struct UiVertex {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/render/mod.rs:603]: If this is pub, we should add doc strings to both of these :)

**Comment by ickshonpe** [comment]
I'm not certain, but I think having access to these fields probably won't help that much and that you'd need to create your own vertex type.

Creating a plugin built on top of https://github.com/bevyengine/bevy/pull/9506 might work.

I've also been working on my corner radius PR today, I had trouble adapting the old PR to work with the recent rendering changes, so I've rewritten the whole thing to use instancing. Just going to carry on working until it's done, should have something to release in a few hours hopefully or tomorrow if I can't fix all the bugs.

**Comment by silvestrpredko** [comment]
@ickshonpe That sounds awesome. Could you please describe more why it didn't work? My original plan is to get access to these fields and then apply the shader that you have in your PR on top of it. Basically, Bevy will render the shape and then I will change it in a few items that I want.

P.S
I don't have a lot of experience with shaders. So my plan could be silly for you. Please point me to my mistakes.

**Comment by ickshonpe** [comment]
I'm not sure what the problems were, it's more than a week since I last worked on the old version. I just decided I was getting nowhere and needed a better understanding of Bevy's rendering internals before I could continue. And after looking at everything and doing some research, instancing seems like the natural approach.

**Comment by silvestrpredko** [comment]
I closed this PR, and during my research, I understood that it didn't help to achieve my purpose.

**Pull request closed without merging**